### DISCUSSION TITLE
How to set Request Timeout in SendHttpRequest in Elsa 3.x

### DISCUSSION THREAD

**Discussion by hmorovati** [elsa-workflows/elsa-core#5649] [Q&A]
Hi
I want to send a http request using SendHttpRequest but it takes a long time and the request is timed out in 100 seconds. I could change the request time out in Elsa Source in SendHttpRequestBase.cs in TrySendAsync but it's not good when deploying the project. 

So could you help me with that?

**Comment by glime-ai** [comment]
To change the timeout for the `SendHttpRequest` activity in Elsa without modifying the source code, you can configure the `HttpClient` used by Elsa when setting up your services. This can be done in the `Startup.cs` or `Program.cs` file of your ASP.NET Core application, depending on the version you are using.

For example, you can configure the `HttpClient` timeout like this:

```csharp
services.AddElsa(elsa => elsa
    .AddHttpActivities(options => options
        .ConfigureHttpClient(client => client.Timeout = TimeSpan.FromMinutes(5)) // Set the desired timeout
    )
);
```

This code snippet sets the timeout for all HTTP requests made by the `SendHttpRequest` activity to 5 minutes. You can adjust the `TimeSpan` to the desired timeout duration.

Remember to place this configuration code in the appropriate place in your application's startup sequence, where you are configuring Elsa services.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by hmorovati** [comment]
It doesn't work in Elsa 3.x
